src/DebugSystem.cpp

```diff
@@ -88,369 +187,247 @@ DebugSystem::DebugSystem(uint32_t rom_start, uint32_t rom_size, uint32_t ram_sta
     }
 }
 
+
+DebugSystem::DebugSystem(void) :
+    rom_start_(-1)
+  , ram_start_(-1)
+  , rom_size_(0)
+  , ram_size_(0) 
+{
+ 
+  _print_ibus_access = etiss::cfg().get<bool>("DebugSystem::printIbusAccess", false);
+  _print_dbus_access = etiss::cfg().get<bool>("DebugSystem::printDbusAccess", false);
+  _print_dbgbus_access = etiss::cfg().get<bool>("DebugSystem::printDbgbusAccess", false);
+  _print_to_file = etiss::cfg().get<bool>("DebugSystem::printToFile", false);
+  message_max_cnt = etiss::cfg().get<int>("DebugSystem::message_max_cnt", 100);
+
+  if (_print_dbus_access)
+  {
+    trace_file_dbus_.open(etiss::cfg().get<std::string>("ETISS::outputPathPrefix", "") + "dBusAccess.csv", std::ios::binary);
+	}
+}
+
+
 etiss::int32 DebugSystem::iread(ETISS_CPU *, etiss::uint64 addr, etiss::uint32 len)
 {
-    if (addr >= _rom_start && addr < _rom_start + rom_mem.size())
-    {
-        return RETURNCODE::NOERROR;
-    }
+  int i_seg = 0;
+  int n_segs = msegs_.size();
+  for(i_seg = 0; n_segs; ++i_seg){
+    if(msegs_[i_seg]->addr_in_range(addr)) break;
```

Comment:
for large number of sections msegs_ should probably be a map, so we can search more efficiently, but the current code should actually be more efficient for the typical two segment binaries we have right now